src/quotamodel.c

```diff
@@ -1515,10 +1540,31 @@ do_load_quotas(void)
 		}
 		else
 		{
-			update_limit_for_quota(quota_limit_mb * (1 << 20), segratio, quotaType, (Oid[]){targetOid, spcOid});
+			if (SearchSysCacheExists1(TABLESPACEOID, ObjectIdGetDatum(spcOid)))
+				update_limit_for_quota(quota_limit_mb * (1 << 20), segratio, quotaType, (Oid[]){targetOid, spcOid});
+			else
+				cleanConfigTables = true;
 		}
 	}
 
+	if (cleanConfigTables)
```

Comment:
Why not do this when deleting an entry from the `quota_info_map` hashmap?

Reply:
I think it is better to don't add an entry to `quota_info_map`

Reply:
But what if the entry is already there?